ipaclient/install/client.py

```diff
@@ -2390,39 +2452,13 @@ def _install(options):
         tasks.backup_hostname(fstore, statestore)
         tasks.set_hostname(options.hostname)
 
-    ntp_srv_servers = []
-    if not options.on_master and options.conf_ntp:
-        # Attempt to sync time with IPA server.
-        # If we're skipping NTP configuration, we also skip the time sync here.
-        # We assume that NTP servers are discoverable through SRV records
-        # in the DNS.
-        # If that fails, we try to sync directly with IPA server,
-        # assuming it runs NTP
-        logger.info('Synchronizing time with KDC...')
-        ds = ipadiscovery.IPADiscovery()
-        ntp_srv_servers = ds.ipadns_search_srv(cli_domain, '_ntp._udp',
-                                               None, break_on_first=False)
-        synced_ntp = False
-        ntp_servers = ntp_srv_servers
-
-        # use user specified NTP servers if there are any
-        if options.ntp_servers:
-            ntp_servers = options.ntp_servers
-
-        for s in ntp_servers:
-            synced_ntp = ntpconf.synconce_ntp(s, options.debug)
-            if synced_ntp:
-                break
-
-        if not synced_ntp and not options.ntp_servers:
-            synced_ntp = ntpconf.synconce_ntp(cli_server[0], options.debug)
-        if not synced_ntp:
-            logger.warning(
-                "Unable to sync time with NTP "
-                "server, assuming the time is in sync. Please check "
-                "that 123 UDP port is opened.")
-    else:
-        logger.info('Skipping synchronizing time with NTP server.')
+    if options.conf_ntp:
+        # Attempt to sync time with NTP server (chrony).
+        sync_time(options, fstore, statestore)
+    elif not options.on_master:
+        # If we're on master skipping the time sync here because it was done
+        # in ipa-server-install
+        logger.info('Skipping synchronizing time with chrony server.')
```

Comment:
:thinking: hmmm first there should not be:
```python
elif not options.on_master:
````
but:
```python
 elif options.on_master:
```
But yes, there might be message to include information. 
I may put sync into separated method if it is what will suits us because it is part of configuration of chrony in method `sync_time()` i could rename it to conf_time() and extract time synchronization attempt. Should I?